### DISCUSSION TITLE
Add boofcv features to Photoslide

### DISCUSSION THREAD

**Discussion by lanthale** [lanthale/PhotoSlide#12] [Ideas]
Which features from boofcv should be added to Photoslide.

Please share your ideas.

**Comment by lessthanoptimal** [comment]
Was creating issues with ideas then noticed this discussion! First time I've used Discussions on GitHub.  So the two "easy" ideas I have right now are

1. Looking up images with similar color histograms. This works surprisingly well for finding related images.
2. Automatic scanning for QR codes. Not sure what this would be used for, but could fill a niche use case
3. Adding image recognition when it's finished. This would find images that were taken of the same scene but different points of view

There is also a lot of 3D photogrametry that could be done automatically but that's probably out of scope for now.

**Reply by lanthale** [reply]
Thanks. 
Number 1 and number 3 is extremely interesting to add as a feature. For the QR code I do not know what the use case should be, but must think about it.

**Reply by lessthanoptimal** [reply]
Let me know if you want to start on the first idea. I'm busy making `#3` a reality so can't help out on the coding side but I can make a bunch of suggestions for how it could be implemented and small tweaks to the code towards the end.

**Reply by lanthale** [reply]
Yes I will do. Have to sort out the dependency management first for the faces detection
